### PR TITLE
Enforce use of file extensions in imports

### LINKED ISSUE #2644: lit-html package with type="module" uses extensionless imports in /directive.d.ts causing typescript to fail.
Note the differences in https://unpkg.com/browse/lit-html@2.2.1/directive.d.ts
vs https://unpkg.com/browse/lit@2.2.1/directive.d.ts

the lit directive.d.ts specifies an extension.
the lit-html directive.d.ts does not, which causes the typescript transpiler to fail!

note: https://unpkg.com/browse/lit-html@2.2.1/package.json specifies itself as "type": module which is why extensionless imports fail.

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes #2644 and adds an eslint plugin to enforce extensions going forward.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -9,11 +9,16 @@
...
+  "plugins": [
+    "eslint-plugin-import",
+    "eslint-plugin-no-only-tests",
+    "@typescript-eslint"
+  ],
   "env": {
     "browser": true
   },
   "rules": {
+    "import/extensions": ["error", "always", {"mjs": "never"}],
```
Comment (justinfagnani) [inline .eslintrc.json:21]: The `.mjs` exclusion is for imports of `uvu/assert`. The TypeScript version we're using doesn't resolve `uvu/assert.mjs`. That might be fixed in a newer version.

File: .eslintignore-sync [hunk 2]
```diff
@@ -6,3 +6,4 @@ packages/{*,labs/*}/{.gitignore,.eslintignore}
 
 [inline]
 packages/localize/testdata/
+packages/localize/examples/
```
Comment (justinfagnani) [inline .eslintignore-sync:9]: This wasn't ignored before and the examples were causing all kinds of lint errors. Are we correctly running lint on CI? @aomarks

File: .eslintrc.json (hunk 1 above)
Comment (justinfagnani) [reply .eslintrc.json:21]: Weird... locally this fixes the problems with `uvu/assert`. On CI it doesn't. 🤷‍♂️

Comment (rictic) [reply .eslintrc.json:21]: I can repro the issue locally by doing:

```
git clone git@github.com:lit/lit.git
git checkout extensions
npm ci
npm run lint
```

**Review by rictic — APPROVED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (justinfagnani) [reply .eslintrc.json]: Well, I couldn't get whatever was causing things to pass locally to work on CI, so I just eslint-ignore'ed those imports.

**Pull request merged**